components/textbox/textbox.tsx

```diff
@@ -66,257 +67,277 @@ type Props = {
-                        channelId: this.props.channelId,
-                        autocompleteUsersInChannel: (prefix: string) => this.props.actions.autocompleteUsersInChannel(prefix, this.props.channelId),
-                        useChannelMentions: this.props.useChannelMentions,
-                        autocompleteGroups: this.props.autocompleteGroups,
-                        searchAssociatedGroupsForReference: (prefix: string) => this.props.actions.searchAssociatedGroupsForReference(prefix, this.props.currentTeamId, this.props.channelId),
-                        priorityProfiles: this.props.priorityProfiles,
+                        currentUserId,
+                        channelId,
+                        autocompleteUsersInChannel: (prefix: string) => actions.autocompleteUsersInChannel(prefix, channelId),
+                        useChannelMentions,
+                        autocompleteGroups,
+                        searchAssociatedGroupsForReference: (prefix: string) => actions.searchAssociatedGroupsForReference(prefix, currentTeamId, channelId),
+                        priorityProfiles,
                     });
                 }
             }
         }
 
-        if (this.props.channelId !== prevProps.channelId ||
-            this.props.currentTeamId !== prevProps.currentTeamId ||
-            this.props.rootId !== prevProps.rootId) {
+        if (channelId !== prevProps.channelId ||
+            currentTeamId !== prevProps.currentTeamId ||
+            rootId !== prevProps.rootId) {
             // Update channel id for CommandProvider and AppCommandProvider.
-            for (const provider of this.suggestionProviders) {
+            for (const provider of suggestionProviders) {
                 if (provider instanceof CommandProvider) {
                     provider.setProps({
-                        teamId: this.props.currentTeamId,
-                        channelId: this.props.channelId,
-                        rootId: this.props.rootId,
+                        teamId: currentTeamId,
+                        channelId,
+                        rootId,
                     });
                 }
                 if (provider instanceof AppCommandProvider) {
                     provider.setProps({
-                        teamId: this.props.currentTeamId,
-                        channelId: this.props.channelId,
-                        rootId: this.props.rootId,
+                        teamId: currentTeamId,
+                        channelId,
+                        rootId,
                     });
                 }
             }
         }
 
-        if (prevProps.value !== this.props.value) {
-            this.checkMessageLength(this.props.value);
-        }
-    }
-
-    componentDidUpdate(prevProps: Props) {
-        if (!prevProps.preview && this.props.preview) {
-            this.preview.current?.focus();
+        if (prevProps.value !== value) {
+            checkMessageLength(value);
```

Comment:
Outside of the above suggestions, I think we'll generally want to break `componentDidUpdate` and such into separate `useEffect` hooks for each task that was done on update rather than having just one big one. Multiple smaller ones would run less often than one big one, and it'd help with keeping logic separated

Reply:
Oh `checkMessageLength`  was part of `this.updateSuggestions` before. Weird